manager/config/config.go

```diff
@@ -0,0 +1,60 @@
+package config
+
+import (
+	"time"
+
+	"github.com/flyteorg/flytestdlib/config"
+)
+
+//go:generate pflags Config --default-var=DefaultConfig
+
+var (
+	DefaultConfig = &Config{
+		PodApplication:       "flytepropeller",
+		PodNamespace:         "flyte",
+		PodTemplateName:      "flytepropeller-template",
+		PodTemplateNamespace: "flyte",
+		ScanInterval: config.Duration{
+			Duration: 10 * time.Second,
+		},
+		ShardConfig: ShardConfig{
+			Type:                   "hash",
+			EnableUncoveredReplica: false,
+			PodCount:               3,
+		},
+	}
+
+	configSection = config.MustRegisterSection("manager", DefaultConfig)
+)
+
+type ShardType = string
+
+const (
+	DomainShardType  ShardType = "domain"
+	ProjectShardType ShardType = "project"
+	HashShardType    ShardType = "hash"
+)
+
+type ReplicaConfig struct {
+	Entities []string `json:"entities" pflag:",The list of entities to be managed"`
+}
+
+type ShardConfig struct {
+	Type                   ShardType       `json:"type" pflag:"\"hash\",Shard implementation to use"`
+	EnableUncoveredReplica bool            `json:"enable-uncovered-replica" pflag:"\"true\",Start a replica to process FlyteWorkflows that are uncovered by the configured shard label selectors"`
+	Replicas               []ReplicaConfig `json:"replicas" pflag:"-"`
```

Comment:
what is difference between replicas and podCount?

isnt podCount like the replica?

Reply:
podCount is used with hashing, replicas (may be terribly named) but refers to a collection of identifiers (e.x. projects / domains) for use with those sharding mechanisms ... do we add separate fields for ProjectReplicas and DomainReplicas? I tried to adhere to the current config setups, but perhaps this case doesn't fit well?